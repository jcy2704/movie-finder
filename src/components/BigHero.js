import React from 'react';
import PropTypes from 'prop-types';
import env from 'react-dotenv';
import '../styles/bighero/BigHero.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar, faPlay } from '@fortawesome/free-solid-svg-icons';

const BigHero = ({
  title, genres, description, backDrop, poster, rating, home, video,
}) => (
  <div className={`position-relative hero ${home ? 'home-screen' : ''}`}>
    <div className="hero-img">
      <img className="w-100 backdrop" src={`${env.IMG_URL}${backDrop}`} alt="BackDrop" />
    </div>

    <div className="position-absolute info-cont d-flex">
      <img src={`${env.POSTER_URL}${poster}`} alt="Poster" />

      <div className="w-75 text-justify info">
        <h1 className="title">{title}</h1>

        <div className="d-flex genres">
          {genres.map(genre => <p key={genre}>{genre}</p>)}
          <p className="rating">
            <FontAwesomeIcon className="rating-star" icon={faStar} />
            {rating}
          </p>
        </div>

        <p className="desc">{description}</p>

        <div className="trailer-cont w-25">
          <a href={`https://www.youtube.com/watch?v=${video.key}`} className="trailer d-flex align-items-center" target="_blank" rel="noreferrer">
            <FontAwesomeIcon className="play-icon" icon={faPlay} />
            Trailer
          </a>
        </div>
      </div>
    </div>
  </div>
);

BigHero.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  backDrop: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  rating: PropTypes.number.isRequired,
  home: PropTypes.bool.isRequired,
  video: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
};

export default BigHero;
