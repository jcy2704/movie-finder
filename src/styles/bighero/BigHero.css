@font-face {
  font-family: 'SF-Pro';
  src: url("../../assets/fonts/SF-Pro-Text-Regular.otf") format("opentype");
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'SF-Pro';
  src: url("../../assets/fonts/SF-Pro-Text-Medium.otf") format("opentype");
  font-weight: 500;
  font-style: normal;
}

@font-face {
  font-family: 'SF-Pro';
  src: url("../../assets/fonts/SF-Pro-Text-Semibold.otf") format("opentype");
  font-weight: 600;
  font-style: normal;
}

html {
  font-size: 100%;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

*,
*::before,
*::after {
  -webkit-box-sizing: inherit;
          box-sizing: inherit;
  -webkit-margin-before: 0;
          margin-block-start: 0;
  -webkit-margin-after: 0;
          margin-block-end: 0;
  margin: 0;
  padding: 0;
  font-family: "SF-Pro", sans-serif;
  color: #fff;
}

body {
  background-color: #222;
}

a {
  text-decoration: none;
  color: inherit;
}

a:hover {
  text-decoration: none;
  color: inherit;
}

.home-screen .backdrop {
  height: 100vh;
}

.hero-img {
  position: relative;
}

.hero-img::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-image: -webkit-gradient(linear, left top, left bottom, from(rgba(0, 0, 0, 0.514)), to(black));
  background-image: linear-gradient(to bottom, rgba(0, 0, 0, 0.514), black);
  opacity: 0.8;
}

.info-cont {
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  width: 85%;
  height: 70%;
  margin: auto;
}

.info-cont img {
  border-radius: 30px;
  margin-right: 30px;
}

.info {
  padding: 30px;
  padding-right: 0;
}

.info .title {
  font-weight: 600;
  font-size: 3em;
}

.info .desc {
  margin-top: 40px;
  margin-bottom: 50px;
  font-size: 1.5em;
}

.genres p {
  color: #ebb52a;
  padding-right: 15px;
  font-size: 1.1em;
}

.genres .rating {
  color: white;
}

.genres .rating .rating-star {
  margin: 0 7px 0 0;
}

.genres .rating .rating-star path {
  color: #ebb52a;
}

.trailer-cont {
  background-color: #ebb52a;
  font-size: 1.25em;
  font-weight: 500;
  color: #222;
  border-radius: 10px;
}

.trailer-cont .trailer {
  padding: 12px 25px;
  cursor: pointer;
}

.trailer-cont .trailer .play-icon {
  margin-right: 15px;
}

.trailer-cont .trailer .play-icon path {
  color: #222;
}
/*# sourceMappingURL=BigHero.css.map */