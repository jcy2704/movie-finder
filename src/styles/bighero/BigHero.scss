@import '../globals';

.home-screen {
  .backdrop {
    height: 100vh;
  }
}

.hero-img {
  position: relative;

  &::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: linear-gradient(to bottom, rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.815));
    opacity: 0.8;
  }
}

.info-cont {
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  width: 85%;
  height: 70%;
  margin: auto;

  img {
    border-radius: 30px;
    margin-right: 30px;
    height: 100%;
  }
}

.info {
  padding: 0 30px;
  padding-right: 0;

  .title {
    font-weight: 600;
    font-size: 3em;
  }

  .desc {
    margin-top: 40px;
    margin-bottom: 55px;
    font-size: 1.5em;
  }
}

.genres {
  p {
    color: $yellow;
    padding-right: 15px;
    font-size: 1.1em;
  }

  .rating {
    color: white;

    .rating-star {
      margin: 0 7px 0 0;

      path {
        color: $yellow;
      }
    }
  }

  button {
    @extend p;

    padding: 0;
  }
}

.trailer-cont {
  background-color: $yellow;
  font-size: 1.25em;
  color: $black;
  border-radius: 10px;

  .trailer {
    font-weight: 500;
    padding: 12px 25px;
    cursor: pointer;

    .play-icon {
      margin-right: 15px;

      path {
        color: $black;
      }
    }
  }
}
