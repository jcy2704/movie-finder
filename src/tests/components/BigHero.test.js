/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import BigHero from '../../components/BigHero';
import GenreList from '../../helpers/genres/genre';

const movie = {
  backdrop_path: '/inJjDhCjfhh3RtrJWBmmDqeuSYC.jpg',
  genre_ids: [
    28,
    878,
  ],
  id: 399566,
  overview: 'In a time when monsters walk the Earth, humanityâ€™s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.',
  poster_path: '/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg',
  title: 'Godzilla vs. Kong',
  vote_average: 8.4,
  video: {},
};

it('Renders Big hero component', () => {
  const popular = renderer
    .create(
      <BigHero title={movie.title} backDrop={movie.backdrop_path} description={movie.overview} poster={movie.poster_path} rating={movie.vote_average} genres={GenreList(movie.genre_ids)} video={movie.video} />,
    )
    .toJSON();

  expect(popular).toMatchSnapshot();
});
