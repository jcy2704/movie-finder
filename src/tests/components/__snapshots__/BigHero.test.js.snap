// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Big hero component 1`] = `
<div
  className="position-relative hero  "
>
  <div
    className="hero-img"
  >
    <img
      alt="BackDrop"
      className="w-100 backdrop"
      src="https://image.tmdb.org/t/p/original/inJjDhCjfhh3RtrJWBmmDqeuSYC.jpg"
    />
  </div>
  <div
    className="position-absolute info-cont d-flex"
  >
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w342/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
    />
    <div
      className="w-75 text-justify info"
    >
      <h1
        className="title"
      >
        Godzilla vs. Kong
      </h1>
      <div
        className="d-flex genres"
      >
        <p>
          <button
            name="Action"
            type="button"
          >
            Action
          </button>
        </p>
        <p>
          <button
            name="Science Fiction"
            type="button"
          >
            Science Fiction
          </button>
        </p>
        <p
          className="rating"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 rating-star"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          8.4
        </p>
      </div>
      <p
        className="desc"
      >
        In a time when monsters walk the Earth, humanityâ€™s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.
      </p>
      <div
        className="trailer-cont w-25"
      >
        <a
          className="trailer d-flex align-items-center"
          href="https://www.youtube.com/watch?v=undefined"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play fa-w-14 play-icon"
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Trailer
        </a>
      </div>
    </div>
  </div>
</div>
`;
